src/parser/Literal.cpp

```diff
@@ -149,4 +149,19 @@ void Literal::setSubstr(std::size_t start, std::size_t length) {
 // __________________________________________
 void Literal::removeDatatypeOrLanguageTag() { content_.erase(beginOfSuffix_); }
 
+// __________________________________________
+void Literal::replaceContent(const std::string& newContent) {
+  std::size_t minLength = std::min(beginOfSuffix_ - 2, newContent.size());
+  for (std::size_t i = 0; i < minLength; ++i) {
+    content_[i + 1] = newContent[i];
+  }
```

Comment:
this is 
```suggestion
  ql::ranges::copy(newContent.substr(0, minLength), content_.begin() + 1);
```
Use algorithms wherever possible.